base.py

```diff
@@ -423,15 +423,14 @@ def _denormalize_quote_join(self, *idents):
         return '.'.join(
             self.identifier_preparer._quote_free_identifiers(*idents))
 
-    @reflection.cache
-    def _current_database_schema(self, connection, **_):
-        row = connection.execute(
-            "SELECT /* sqlalchemy:_current_database_schema */ "
-            "CURRENT_DATABASE(), CURRENT_SCHEMA()").fetchone()
-        return self.normalize_name(row[0]), self.normalize_name(row[1])
-
-    def _get_default_schema_name(self, connection, **kw):
-        _, current_schema = self._current_database_schema(connection, **kw)
+    def _current_database_schema(self, connection):
+        return (
+            self.normalize_name(connection.connect().connection.database),
+            self.normalize_name(connection.connect().connection.schema))
```

Comment:
Use cached current database and schema.

Those attributes are updated on every query, so they should be up to date.